test/Azure.Functions.Cli.Tests/E2E/AzureResourceManagers/BaseAzureResourceManager.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Azure.Functions.Cli.Common;
+using System;
+
+namespace Azure.Functions.Cli.Tests.E2E.AzureResourceManagers
+{
+    public class BaseAzureResourceManager: IDisposable
```

Comment:
You can probably mark this and `MultiOsResourcesManager` class as `abstract`. Or remove the `throw new NotImplementedException();` in `Cleanup` here. I say this as creating an instance of this and disposing would throw an exception.
And it seems like you'd not want to instantiate this directly, so maybe make this abstract?